libnetdata/threads/threads.c

```diff
@@ -109,6 +109,33 @@ static void thread_cleanup(void *ptr) {
     netdata_thread = NULL;
 }
 
+static void thread_set_name(NETDATA_THREAD *nt) {
+
+    if (nt->tag) {
+        char *threadname = strdupz(nt->tag);
+        int ret = 0;
+
+        if (threadname) {
+            // Name is limited to 16 chars
+            threadname[15] = 0;
+#if defined(__FreeBSD__)
+            ret = pthread_set_name_np(pthread_self(), threadname);
```

Comment:
The `pthread_set_name_np()` function has a `void` return type so you should remove `ret = ` because it breaks compilation in FreeBSD.

Reply:
Done. Seems pthread on apple is returning int too.

It's a shame travis doesn't provide freebsd env for testing :(

Reply:
At the moment I'll have to manually test freebsd and ask somebody else to do the same for OS X.